scripts: board_v1_to_v2: License&copyright + SOC_SERIES fix

erwango opened the pull request:
This PR is adding 2 changes:

- When creating file Kconfig.<board> a copyright is required as otherwise an issue will be reported by CI scan code tool.
Always add License (SPDX) and add the option to put a copyright.

- Don't add select CONFIG_SOC_SERIES_FOO as part of board description. It is done at soc level.

erwango committed:
scripts: board_v1_to_v2: Add License + copyright

When creating file Kconfig.<board>, get the copyright from
old Kconfig.board file and use it in the new file.
Add the License as well.

Signed-off-by: Erwan Gouriou <erwan.gouriou@st.com>

gmarull commented on scripts/utils/board_v1_to_v2.py:
```diff
@@ -157,11 +158,15 @@ def board_v1_to_v2(board_root, board, new_board, group, vendor, soc, variants):
-    selects = "\n\t" + "\n\t".join(["select " + setting for setting in board_soc_settings])
+    header = "# SPDX-License-Identifier: Apache-2.0\n"
+    if copyright is not None:
+        header = header + "# Copyright (c) 2024 " + copyright + "\n"
```
hardcoding year isn't likely a good idea (although... I hope we have finished by 2025)

erwango replied:
Making it configurable sounded a bit over-engineered given the condition of usage.

erwango replied:
Resolved as I'm now retrieving previous Copyright.

gmarull commented on scripts/utils/board_v1_to_v2.py:
```diff
@@ -157,11 +158,15 @@ def board_v1_to_v2(board_root, board, new_board, group, vendor, soc, variants):
     print(f"Creating or updating Kconfig.{new_board}...")
     new_board_kconfig_file = new_board_path / f"Kconfig.{new_board}"
-    selects = "\n\t" + "\n\t".join(["select " + setting for setting in board_soc_settings])
+    header = "# SPDX-License-Identifier: Apache-2.0\n"
```
I wonder if we should just re-use copyright from the deleted "Kconfig.board"

erwango replied:
You're right. I'll have another shot.

erwango replied:
@gmarull What if we just:
- move and rename the existing Kconfig.board
- drop the "bool"
- s/depends on/select

gmarull replied:
some selects will need to be removed (e.g. all SOC_SERIES*), I need to update the script for that. So not sure if easier to update the file or regenerate it.

erwango replied:
On some boards, we'd need to remove the help as well indeed. See https://github.com/zephyrproject-rtos/zephyr/pull/67737/files

gmarull replied:
so then I'd just grab the copyright and create a new file as we do now

erwango replied:
> some selects will need to be removed (e.g. all SOC_SERIES*), I need to update the script for that. So not sure if easier to update the file or regenerate it.

Do you think about something like follows ?
```
-            m = re.match(r"^CONFIG_(SOC_[A-Z0-9_]+).*$", line)
+            m = re.match(r"^CONFIG_(SOC_(?!SERIES_)[A-Z0-9_]+).*$", line)
```

gmarull replied:
yes, willing to add a patch :-)?

erwango replied:
Done ;)

gmarull commented on scripts/utils/board_v1_to_v2.py:
```diff
@@ -156,19 +156,27 @@ def board_v1_to_v2(board_root, board, new_board, group, vendor, soc, variants):
     new_board_kconfig_file = new_board_path / f"Kconfig.{new_board}"
-    selects = "\n\t" + "\n\t".join(["select " + setting for setting in board_soc_settings])
+    header = "# SPDX-License-Identifier: Apache-2.0\n"
+    if copyright is not None:
+        header = header + copyright
```
shouldn't this be 
```suggestion
    if copyright is not None:
        header = copyright + header
```
?

erwango replied:
Right.

erwango committed:
scripts: board_v1_to_v2: Don't add select CONFIG_SOC_SERIES_FOO

SOC_SERIES_FOO selection is done at soc/ level it should
not be added as part of board description

Signed-off-by: Erwan Gouriou <erwan.gouriou@st.com>

erwango commented:
Set as DNM as proposed change is not sufficient to drop `CONFIG_SOC_SERIES_STM32F0X` line in board_defconfig file

erwango commented:
@gmarull Hope it is fine now.
